python/vyos/ifconfig/interface.py

```diff
@@ -1359,7 +1359,7 @@ def update(self, config):
         self.set_ipv4_source_validation(value)
 
         # Only change IPv6 parameters if IPv6 was not explicitly disabled
-        if sysctl_read('net.ipv6.conf.all.disable_ipv6') == '0':
+        if is_ipv6_enabled():
             # IPv6 forwarding
             tmp = dict_search('ipv6.disable_forwarding', config)
```

Comment:
Please note that there are two different commands with different intended effects.

`set system ipv6 disable-forwarding` is supposed to _only_ disable IPv6 packet routing. It isn't for disabling IPv6 completely, the system may still act as an IPv6 web proxy, DNS resolver, or perform some other network service that doesn't involve forwarding unprocessed packets between interfaces.

It's `set system ipv6 disable` that is meant to make the system IPv4-only: not just no forwarding, but also no IPv6 loopback, no link-locals, no other traces of IPv6.

Well, that's the uncontroversial part of my comment, just a correction.

**However!** One reason I put a sysctl read there was that IPv6 may potentially be disabled by something else: some cloud-init stuff, kernel command line, or a user who loves messing up with sysctl... I'm not sure if that idea has unfortunate implications that just reading from the config would avoid, but I think this issue does deserve more discussion.